tests/test_timeouts.py

```diff
@@ -29,6 +29,16 @@
             await client.get("http://10.255.255.1/")
 
 
+@pytest.mark.asyncio
+async def test_connect_timeout_on_request_call(server):
+    timeout = TimeoutConfig(connect_timeout=0.0001)
+
+    async with Client() as client:
+        with pytest.raises(ConnectTimeout):
+            # See https://stackoverflow.com/questions/100841/
+            await client.get("http://10.255.255.1/", timeout=timeout)
```

Comment:
Let's pass on this for now - some stuff still to be re-jigged around config so maybe it'll fall away anyhows.